detox/src/artifacts/screenshot/SimulatorScreenshotPlugin.js

```diff
@@ -36,16 +37,31 @@ class SimulatorScreenshotPlugin extends ScreenshotArtifactPlugin {
     }
   }
 
-  _onInvokeFailure({ params }) {
-    const { visibilityFailingScreenshotsURL, visibilityFailingRectsURL } = params;
+  async onBeforeUninstallApp(event) {
+    const snapshots = [
+      ...Object.values(this.snapshots.fromTest),
+      ...Object.values(this.snapshots.fromSession)
+    ];
 
-    this._registerSnapshot('visibilityFailingScreenshots/', new FileArtifact({
-      temporaryPath: visibilityFailingScreenshotsURL
-    }));
+    await Promise.all(snapshots.map(s => s && s.relocate()));
+    await super.onBeforeUninstallApp(event);
+  }
+
+  async _onInvokeFailure({ params }) {
+    const artifacts = {
+      visibilityFailingScreenshots: params.visibilityFailingScreenshotsURL,
+      visibilityFailingRects: params.visibilityFailingRectsURL,
+    };
 
-    this._registerSnapshot('visibilityFailingRects/', new FileArtifact({
-      temporaryPath: visibilityFailingRectsURL
-    }));
+    for (const [key, artifactPath] of Object.entries(artifacts)) {
+      if (!artifactPath || await fs.isDirEmpty(artifactPath)) {
```

Comment:
For some reason, I can see dirs sent, but they are empty in fact. Maybe a bit later, we should look into it on iOS.